### PR TITLE
enable configuring 'single-node' variant

### PR DESCRIPTION
**Pull request by osherdp**
Using 1 master+worker node in SNO (Single Node Openshift) cluster installation.

### CODE REVIEW

**Comment by pierreprinetti** [comment]
You might need to rebase and `make update` to make the validator happy

**Comment by osherdp** [comment]
@pierreprinetti ok
let me just run some more tests, I want to make sure of something

**Comment by osherdp** [comment]
@pierreprinetti can you take another look?
thanks!

**Comment by osherdp** [comment]
@smarterclayton @romfreiman be assure that failure of e2e-aws-sno is an expected failure

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -453,10 +453,16 @@ objects:
         mkdir -p ~/.ssh
         cp "${SSH_PRIV_KEY_PATH}" ~/.ssh/
 
+        masters=3
+        if has_variant "sno" ;then
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml:457]: Let's make the variant `single-node`.  `sno` doesn't mean anything and we have characters to spare.

Comment (osherdp) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: OK, done

File: core-services/template-deprecation/_allowlist.yaml [hunk 2]
```diff
@@ -736,6 +736,9 @@ templates:
         pull-ci-openshift-installer-master-e2e-aws-rhel8:
           generated: false
           kind: presubmit
+        pull-ci-openshift-installer-master-e2e-aws-sno:
```
Comment (smarterclayton) [inline core-services/template-deprecation/_allowlist.yaml:739]: call this `-aws-single-node`

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 3]
```diff
@@ -507,6 +507,89 @@ presubmits:
...
+    branches:
+    - master
+    cluster: build02
+    context: ci/prow/e2e-aws-sno
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-installer-master-e2e-aws-sno
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:521]: `-aws-single-node`

Comment (osherdp) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: Done, everywhere

File: core-services/template-deprecation/_allowlist.yaml (hunk 2 above)
Comment (osherdp) [reply core-services/template-deprecation/_allowlist.yaml]: Done

**Comment by osherdp** [comment]
@smarterclayton @romfreiman all should be good now
tell me if anything else needed

**Pull request merged**